### PR TITLE
bytestring normalization

### PR DESCRIPTION
**Pull request by jrobeson**
Make sure many/most strings that touch the filesystem are
converted explicitly to bytestrings rather than implictly.

Definitely review to make sure I used the right constructs/functions

### CODE REVIEW

**Comment by jrobeson** [comment]
I have other related stuff in my py3 branch, but I'm still noodling on how to handle them.

File: beets/importer.py [hunk 1]
```diff
@@ -1483,7 +1483,8 @@ def albums_in_dir(path):
         # named in this way.
         start_collapsing = False
         for marker in MULTIDISC_MARKERS:
-            marker_pat = re.compile(MULTIDISC_PAT_FMT % marker, re.I)
+            p = MULTIDISC_PAT_FMT.replace(b'%s', marker)
```
Comment (sampsyo) [inline beets/importer.py]: This line could use a comment about why we're using .replace instead of %.

File: test/test_library.py [hunk 2]
```diff
@@ -883,7 +883,8 @@ def setUp(self):
 
     def test_art_filename_respects_setting(self):
         art = self.ai.art_destination('something.jpg')
-        self.assertTrue('%sartimage.jpg' % util.PATH_SEP in art)
+        new_art = bytestring_path('%sartimage.jpg' % os.path.sep)
+        self.assertTrue(new_art in art)
```
Comment (sampsyo) [inline test/test_library.py]: Might this have clobbered a switch to `util.PATH_SEP`?

**Comment by sampsyo** [comment]
Whoa! This is a *truly* tremendous amount of work! You have my sincere gratitude for combing through all of these inconsistencies. This will be great for the project long-term.

I reviewed everything and only found two small changes to suggest. After that, please feel free to merge.

File: test/test_library.py (hunk 2 above)
Comment (jrobeson) [inline test/test_library.py]: `util.PATH_SEP` is a bytestring, so that wouldn't work here

File: beets/importer.py (hunk 1 above)
Comment (jrobeson) [inline beets/importer.py]: can do.

**Comment by jrobeson** [comment]
thanks ;)

File: test/test_library.py (hunk 2 above)
Comment (sampsyo) [inline test/test_library.py]: Aha; good catch!

**Pull request merged**